From 03e0037c8ae9842f6dc4fb0cc56424786f04c1f4 Mon Sep 17 00:00:00 2001
From: Rakesh Babu <rsaladi2@marvell.com>
Date: Thu, 25 Mar 2021 15:49:01 +0530
Subject: [PATCH 1335/1921] octeontx2-pf: Fix for ping among LBK pairs.

NPA aura backpressure in aura context is not properly set for NIX1 RX
interface. Because of this, ping among LBK pairs is not working in 98xx.
This patch fixes this.

Fixes: 89d796261c9a ("octeontx2-pf: Set NIX1_RX interface for backpressure")
Change-Id: I408519711f848fc16658e5c58ab76c303cc2b319
Signed-off-by: Rakesh Babu <rsaladi2@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/48636
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/nic/otx2_common.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_common.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_common.c
index 760361dfd220..e6cd6ef8233c 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_common.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_common.c
@@ -1171,9 +1171,13 @@ static int otx2_aura_init(struct otx2_nic *pfvf, int aura_id,
 		 * In the above description 'One bit per NIX-RX' is written
 		 * presumably by mistake in HRM.
 		 */
-		if (pfvf->nix_blkaddr == BLKADDR_NIX1)
+		if (pfvf->nix_blkaddr == BLKADDR_NIX1) {
 			aq->aura.bp_ena = 1;
-		aq->aura.nix0_bpid = pfvf->bpid[0];
+			aq->aura.nix1_bpid = pfvf->bpid[0];
+		} else {
+			aq->aura.nix0_bpid = pfvf->bpid[0];
+		}
+
 		/* Set backpressure level for RQ's Aura */
 		aq->aura.bp = RQ_BP_LVL_AURA;
 	}
-- 
2.31.1

